### DISCUSSION TITLE
Infinite loop with zustand@5.0.0-rc.1

### DISCUSSION THREAD

**Discussion by Romainlg29** [pmndrs/zustand#2716] [Bug report] [answered]
### Bug Description

While updating my zustand librairie to v5, i'm getting some errors that were not here with Zustand 4.5.5.

![image](https://github.com/user-attachments/assets/3cbb9b7d-4d51-45ee-a34f-03c8d251b706)

[Link: Repository with zustand 5-rc](https://github.com/Romainlg29/use-broadcast/pull/16/files)

### Reproduction Link

https://stackblitz.com/~/github.com/Romainlg29/use-broadcast

**Comment by dai-shi** [comment, marked as answer]
Thanks for reporting. Really great to see people try v5-rc.

I think it's the intentional change.
Please check this:
https://github.com/pmndrs/zustand/blob/9e8b7a5eea9c0a9f04dccf9e02fbfe9e1f4fd9e9/docs/migrations/migrating-to-v5.md?plain=1#L92-L129

**Reply by Romainlg29** [reply]
Thank you, it's working well with both `useShallow` and `createWithEqualityFn`.
